lib/reward.ts

```diff
@@ -0,0 +1,71 @@
+import type { Address, PublicClient } from 'viem'
+import { formatEther, encodeAbiParameters} from 'viem';
+import proxyAbi from '@/lib/proxy-abi';
+import { PROXY_CONTRACT_ADDRESS } from '@/lib/config';
+
+
+async function getRewarderAddress(
+  attestationType: string,
+  client: PublicClient,
+): Promise<Address> {
+  const [,rewarderAddress,] = await client.readContract({
+    address: PROXY_CONTRACT_ADDRESS,
+    abi: proxyAbi,
+    functionName: 'attestationTypes',
+    args: [attestationType],
+  }) as [any, Address, any];
+  return rewarderAddress;
+}
+
+
+const rewarderABI = [{
+  "type":"function",
+  "name":"rewardFor",
+  "inputs":[{
+    "name":"attestationData",
+    "type":"bytes",
+  }],
+  "outputs":[{
+    "type":"uint256",
+  }],
+  "stateMutability":"view"
+}]
+
+async function getReward(
+  rewarder: Address,
+  client: PublicClient,
+  data: string
+): Promise<bigint> {
+  const reward = await client.readContract({
+    address: rewarder,
+    abi: rewarderABI,
+    functionName: 'rewardFor',
+    args: [data],
+  }) as bigint;
```

Comment:
when you add `as const` in the reward abi expression, this should be inferred automatically